portfolio/src/main/webapp/script.js

```diff
@@ -39,24 +40,157 @@ function refreshComments() {
   }
 }
 
+/** Obtain comment data from datastore and display them on page */
 function getData() {
   refreshComments();
-  var numContactsToDisplay = document.getElementById('quantity').value;
-  var link = '/data?numContactsToDisplay=' + numContactsToDisplay
+  const numContactsToDisplay = document.getElementById('quantity').value;
+  let link = '/data?numContactsToDisplay=' + numContactsToDisplay;
 
+  // fetch  the data and create a list containing the message of each contact
   fetch(link).then(response => response.json()).then((contacts) => {
+    console.log('Hello World');
+    const infoContainer = document.getElementById('more-info');
     contacts.forEach((contact) => {
-      var currContact = document.createElement('li');
-      var userInfo = document.createTextNode(
+      const currContact = document.createElement('li');
+      const userInfo = document.createTextNode(
           contact.firstName + ' ' + contact.lastName + ' says ' +
           contact.subject);
       currContact.appendChild(userInfo);
-      document.getElementById('more-info').appendChild(currContact);
+      infoContainer.appendChild(currContact);
     });
   });
 }
 
 function deleteData() {
+  // upon request navigate to the delete servlet and remove all comments from
+  // datastore
   const request = new Request('/delete-data', {method: 'POST'});
   fetch(request).then((results) => getData());
 }
+
+/** Allow the user to login, when logged in -> display contact form */
+function userLogin() {
+  fetch('/login').then((response) => response.json()).then((currUser) => {
+    document.getElementById('form-overlay-text').innerHTML =
+        currUser.loginMessage;
+    if (currUser.isLoggedIn) {
+      document.getElementById('form-overlay').style.display = 'block';
+    } else {
+      document.getElementById('form-overlay').style.display = 'none';
+    }
+    // display delete all comments button to one specified user => admin
+    if (currUser.email == 'eretana@google.com') {
```

Comment:
it's bad to hardcode emails like this. it might be better to add an isAdmin variable to UserStatus. we can use this to determine if user is an admin: https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/users/UserService.html#isuseradmin